src/Microsoft.TestPlatform.CrossPlatEngine/Utilities/SourceDetailHelper.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using System.Xml;
+
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Utilities;
+using Microsoft.VisualStudio.TestPlatform.Utilities;
+
+namespace Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Utilities;
+
+internal static class SourceDetailHelper
+{
+    internal static string UpdateRunSettingsFromSourceDetail(string runSettings, SourceDetail sourceDetail)
+    {
+        using var stream = new StringReader(runSettings);
+        var reader = XmlReader.Create(stream, XmlRunSettingsUtilities.ReaderSettings);
```

Comment:
Sorry I missed it earlier, this one also should be disposed

```suggestion
        using var reader = XmlReader.Create(stream, XmlRunSettingsUtilities.ReaderSettings);
```